source/bdDetect.py

```diff
@@ -88,7 +90,8 @@ def __getattr__(attrName: str) -> Any:
 	raise AttributeError(f"module {repr(__name__)} has no attribute {repr(attrName)}")
 
 
-class DeviceMatch(NamedTuple):
+@dataclass
+class DeviceMatch(Sequence):

```

Comment:
On reflection, I wonder whether this change contributes to a solution. Sorry about that. The entire detection system is not made for this kind of stuff.

Reply:
@LeonarddeR 
Thank you for comment.
Do you think that the previous code is not suitable? Should we roll it back?

Reply:
I rollback DeviceMatch code. And make not using that.